### PR TITLE
grpc access log: verify cluster exists and is static

### PR DESCRIPTION
**Pull request by mattklein123**
*Risk Level*: Low 
*Testing*: Unit
*Docs Changes*: N/A
*Release Notes*: N/A

### CODE REVIEW

**Comment by mattklein123** [comment]
@junr03 @ccaraman @lita @rodaine PTAL

File: source/server/config/access_log/grpc_access_log.cc [hunk 1]
```diff
@@ -41,14 +41,20 @@ AccessLog::InstanceSharedPtr HttpGrpcAccessLogFactory::createAccessLogInstance(
   const auto& proto_config = MessageUtil::downcastAndValidate<
       const envoy::api::v2::filter::accesslog::HttpGrpcAccessLogConfig&>(config);
 
+  // TODO(htuch): Support Google gRPC client.
+  const auto cluster_name = proto_config.common_config().grpc_service().envoy_grpc().cluster_name();
+  auto cluster = context.clusterManager().get(cluster_name);
+  if (cluster == nullptr || cluster->info()->addedViaApi()) {
```
Comment (lita) [inline source/server/config/access_log/grpc_access_log.cc]: what is the difference between a static cluster vs a cluster added by api?

Comment (ccaraman) [reply source/server/config/access_log/grpc_access_log.cc]: Static cluster is one defined in the config. A cluster added by api would be in the CDS case.

**Comment by mattklein123** [comment]
Discussed this with @htuch and I think we will move this check into the base AsyncClient for now as I think we want this in all existing use cases.

**Comment by mattklein123** [comment]
Hmm. I just looked and this is not so easy, as we don't create the AsyncClient in this case until we get into TLS code, unlike many of the existing main thread use cases. I think the best I can do is potentially make a utility function for this check? @htuch WDYT?

**Comment by htuch** [comment]
We may have a more fundamental issue here. Are we relying on client
creation to be cheap and potentially reuse an existing connection? The
Google gRPC client creates a new channel each time it instantiates, I'm
wondering if this violates this. Otoh, maybe it does what Envoy does
internally with connection pooling.

**Comment by mattklein123** [comment]
> Are we relying on client creation to be cheap and potentially reuse an existing connection?

In general, yes, but not in the case of access log where we are more careful about TLS data structures and will keep the client around for a long time. There are other cases (rate limit is one for sure) where we create a new client on every request. This is not optimal but works OK with the Envoy gRPC client.

**Comment by mattklein123** [comment]
@htuch I'm inclined to just merge this and we can reconcile the error handling in a follow up PR? WDYT?

**Comment by ramaraochavali** [comment]
@mattklein123 @htuch i was thinking of doing similar validation for metrics service. Should I do the same or wait till it is finalised on how this should be done generically?

**Comment by htuch** [comment]
I have a PR that I will post soon that consolidates allmthis checking, so
no worries, thanks.

**Comment by ramaraochavali** [comment]
Cool.  Thanks

**Comment by ramaraochavali** [comment]
@htuch did you push this static cluster validation?

**Comment by htuch** [comment]
@ramaraochavali yes, this was done in https://github.com/envoyproxy/envoy/pull/2393.

**Comment by ramaraochavali** [comment]
@htuch Great. Thanks.

**Pull request merged**